### PR TITLE
Meeting state slider

### PR DESCRIPTION
**Pull request by MirkoTV**
Added meeting state slider to settings view

### CODE REVIEW

File: src/main/java/org/jala/efeeder/foodmeeting/EditFoodMeetingCommand.java [hunk 1]
```diff
@@ -36,13 +37,19 @@ public Out execute(In parameters) throws Exception {
 		DateTimeFormatter formatter = DateTimeFormat.forPattern("dd MMMM, yyyy HH:mm");
 		DateTime dateTime = formatter.parseDateTime(parameters.getParameter("date") + " " + parameters.getParameter("time"));
 		Timestamp eventDate = new Timestamp(dateTime.getMillis());
+
```
Comment (rayaviri) [inline src/main/java/org/jala/efeeder/foodmeeting/EditFoodMeetingCommand.java]: delete one line

File: src/main/java/org/jala/efeeder/foodmeeting/FoodMeeting.java [hunk 2]
```diff
@@ -47,6 +51,26 @@ public int getWidth() {
 
 		return calculateWidth(days);
 	}
+
```
Comment (rayaviri) [inline src/main/java/org/jala/efeeder/foodmeeting/FoodMeeting.java]: delete one tab

File: src/main/java/org/jala/efeeder/foodmeeting/FoodMeeting.java [hunk 3]
```diff
@@ -47,6 +51,26 @@ public int getWidth() {
...
+		Timestamp now = new Timestamp((new Date()).getTime());
+		
+		if(now.before(this.votingDate)) {
+			status = FoodMeetingStatus.Voting.name();
+		} else if(now.before(this.orderDate)) {
+			status = FoodMeetingStatus.Order.name();
+		} else if(now.before(this.paymentDate)) {
+			status = FoodMeetingStatus.Payment.name();
+		} else if(now.before(this.eventDate))
+		{
```
Comment (rayaviri) [inline src/main/java/org/jala/efeeder/foodmeeting/FoodMeeting.java]: move to up line the {

File: src/main/java/org/jala/efeeder/foodmeeting/FoodMeetingStatus.java [hunk 4]
```diff
@@ -7,5 +7,7 @@
 public enum FoodMeetingStatus {
     Voting,
     Order,
+	Payment,
```
Comment (rayaviri) [inline src/main/java/org/jala/efeeder/foodmeeting/FoodMeetingStatus.java:10]: align all elements

File: src/main/java/org/jala/efeeder/foodmeeting/GetAllMeetingsCommand.java [hunk 5]
```diff
@@ -36,7 +37,8 @@ public Out execute(In parameters) throws Exception {
 		while (resultSet.next()) {
 
 			foodMeetings.add(new FoodMeeting(resultSet.getInt(1), resultSet.getString(2), resultSet.getString(3),
-					resultSet.getString(4), resultSet.getTimestamp(5), resultSet.getTimestamp(6), userManager.getUserById(resultSet.getInt(7))));
+				    resultSet.getTimestamp(4), resultSet.getTimestamp(5), resultSet.getTimestamp(6),
+					resultSet.getTimestamp(7), resultSet.getTimestamp(8), userManager.getUserById(resultSet.getInt(9))));
```
Comment (rayaviri) [inline src/main/java/org/jala/efeeder/foodmeeting/GetAllMeetingsCommand.java]: align

File: src/main/webapp/WEB-INF/tags/template.tag [hunk 6]
```diff
@@ -18,11 +18,12 @@
 		<link href="/assets/css/select2.min.css" rel="stylesheet">
 		<link href="/assets/css/app.css" rel="stylesheet">
 		<link href="/assets/css/foodMeetings.css" rel="stylesheet">
-		<link href="/assets/css/order.css" rel="stylesheet">
+		<link href="/assets/css/order.css" rel="stylesheet">
```
Comment (rayaviri) [inline src/main/webapp/WEB-INF/tags/template.tag]: end  white spaces

File: src/main/webapp/assets/css/nouislider.css [hunk 7]
```diff
@@ -0,0 +1,455 @@
+/*! nouislider - 9.0.0 - 2016-09-29 21:44:02 */
+
+
+/* Functional styling;
+ * These styles are required for noUiSlider to function.
+ * You don't need to change these rules to apply your design.
+ */
+.noUi-target,
+.noUi-target * {
+-webkit-touch-callout: none;
```
Comment (rayaviri) [inline src/main/webapp/assets/css/nouislider.css]: indentation

File: src/main/webapp/assets/css/nouislider.css [hunk 8]
```diff
@@ -0,0 +1,455 @@
...
+	z-index: 1;
+}
+
+
+
+
+
+.noUi-state-tap .noUi-connect,
+.noUi-state-tap .noUi-origin {
+-webkit-transition: top 0.3s, right 0.3s, bottom 0.3s, left 0.3s;
```
Comment (rayaviri) [inline src/main/webapp/assets/css/nouislider.css]: indentation

File: src/main/webapp/assets/css/nouislider.css [hunk 9]
```diff
@@ -0,0 +1,455 @@
...
+	background: #cfd8dc;
+
+	border-radius: 4px;
+	border: 1px solid #D3D3D3;
+	box-shadow: inset 0 1px 1px #F0F0F0, 0 3px 6px -5px #BBB;
+}
+.noUi-connect {
+	background: #3FB8AF;
+	box-shadow: inset 0 0 3px rgba(51,51,51,0.45);
+-webkit-transition: background 450ms;
```
Comment (rayaviri) [inline src/main/webapp/assets/css/nouislider.css]: indentation

File: src/main/webapp/assets/css/nouislider.css [hunk 10]
```diff
@@ -0,0 +1,455 @@
...
+}
+
+
+
+/* Base;
+ *
+ */
+.noUi-pips,
+.noUi-pips * {
+-moz-box-sizing: border-box;
```
Comment (rayaviri) [inline src/main/webapp/assets/css/nouislider.css]: indentation

File: src/main/webapp/assets/css/nouislider.css [hunk 11]
```diff
@@ -0,0 +1,455 @@
...
+	box-sizing: border-box;
+}
+.noUi-pips {
+	position: absolute;
+	color: #9e9e9e;
+	font-family: "Roboto", sans-serif;
+	font-size: 15px;
+    font-weight: 300;
```
Comment (rayaviri) [inline src/main/webapp/assets/css/nouislider.css:179]: align

File: src/main/webapp/assets/css/nouislider.css [hunk 12]
```diff
@@ -0,0 +1,455 @@
...
+ */
+.noUi-marker {
+
+
+	position: absolute;
+	background: #CCC;
+
+}
+.noUi-marker-sub {
+
```
Comment (rayaviri) [inline src/main/webapp/assets/css/nouislider.css]: remove

File: src/main/webapp/assets/css/nouislider.css [hunk 13]
```diff
@@ -0,0 +1,455 @@
...
+	background: #AAA;
+}
+
```
Comment (rayaviri) [inline src/main/webapp/assets/css/nouislider.css:205]: remove lines

File: src/main/webapp/assets/css/nouislider.css [hunk 14]
```diff
@@ -0,0 +1,455 @@
...
+
+
+.noUi-marker-large {
+	background: #AAA;
+
```
Comment (rayaviri) [inline src/main/webapp/assets/css/nouislider.css]: remove

File: src/main/webapp/assets/css/nouislider.css [hunk 15]
```diff
@@ -0,0 +1,455 @@
...
+}
+
```
Comment (rayaviri) [inline src/main/webapp/assets/css/nouislider.css:211]: remove lines

File: src/main/webapp/assets/css/nouislider.css [hunk 16]
```diff
@@ -0,0 +1,455 @@
...
+
+
+/* Horizontal layout;
+ *
+ */
+.noUi-pips-horizontal {
+	padding: 10px 0;
+	height: 80px;
+
```
Comment (rayaviri) [inline src/main/webapp/assets/css/nouislider.css]: you need to fix this file,

File: src/main/webapp/assets/js/settingMeeting.js [hunk 17]
```diff
@@ -8,7 +8,8 @@ var SettingMeeting = function() {
 	var dateField = $('#date-field-id');
 	var timeField = $('#time-field-id');
 	var select = $('select');
-
+	var meetingStateSlider = new MeetingStateSllider();
+
```
Comment (rayaviri) [inline src/main/webapp/assets/js/settingMeeting.js:14]: remove  the spaces or tabs

File: src/main/webapp/assets/js/settingMeeting.js [hunk 18]
```diff
@@ -29,6 +30,8 @@ var SettingMeeting = function() {
 		});
 
 		select.material_select();
+
```
Comment (rayaviri) [inline src/main/webapp/assets/js/settingMeeting.js]: remove spaces or tab

File: src/main/webapp/assets/js/settingMeeting.js [hunk 19]
```diff
@@ -37,7 +40,7 @@ var SettingMeeting = function() {
 		timeField.val(time);
 		var value = moment(dateField.val()).format("D MMMM, YYYY");
 		dateField.val(value);
-	};
+	};
```
Comment (rayaviri) [inline src/main/webapp/assets/js/settingMeeting.js]: remove end characters

File: src/main/webapp/assets/js/settingMeeting.js [hunk 20]
```diff
@@ -47,3 +50,160 @@ var SettingMeeting = function() {
 	};
 
 };
+
+var MeetingStateSllider = function() {
+	var slider = document.getElementById('noUiSlider');
+	var eventDate= moment($("#edit-meeting").data("eventDate"), "YYYY-MM-DD HH:mm:ss");
+	var createdDate = moment($("#edit-meeting").data("createdDate"), "YYYY-MM-DD HH:mm:ss");
+
```
Comment (rayaviri) [inline src/main/webapp/assets/js/settingMeeting.js:80]: remove this characters in all file

File: src/main/webapp/assets/css/nouislider.css (hunk 16 above)
Comment (MirkoTV) [reply src/main/webapp/assets/css/nouislider.css]: yes, It was a third party library, so I didn't notice all those spaces :P, I will fix it right now!

File: src/main/webapp/assets/js/settingMeeting.js (hunk 17 above)
Comment (MirkoTV) [reply src/main/webapp/assets/js/settingMeeting.js:14]: I think this one is necessary since a method is next

**Review by rayaviri — APPROVED** [review]
looks fine, please resolve the conflict before merge your changes

**Pull request merged**